aten/src/ATen/native/nested/cuda/NestedTensorTransformerFunctions.cpp

```diff
@@ -204,5 +207,36 @@ Tensor NestedTensor_to_padded_tensor_cuda(
   }
   return NestedTensor_to_padded_tensor_generic(t, padding, output_size);
 }
+
+Tensor flash_scaled_dot_product_self_attention(
+    const Tensor& qkv,
+    const Tensor& cumulative_sequence_length,
+    const int64_t max_seqlen_batch,
+    double dropout_p,
+    bool causal) {
+#if defined(CUDA_VERSION) && CUDA_VERSION >= 11060 && !defined(_MSC_VER) && \
```

Comment:
Not totally sure what to do here. I have a cmake flag that looks for similar conditions and sets the USE_FLASH_ATTENTION flag to true and if thats the case we will build all the flash_atten sources. However if you were on an machine that has linux and is using cuda 11.6 and is linux then this if_define be true but someone could set USE_FLASH_ATTENTION to FALSE and if thats the case this will result in undefined_symbol for mha_fwd.

Not to sure how to resolve this yet

Reply:
Maybe: Use `USE_FLASH_ATTENTION` here as well. If that flag isn't True, then we're not building with it.

Reply:
ASAI I don't think that cmake Options are available to compilier for macro expansion but I am trying to figure out if there is a different way to set USE_FLASH_ATTENTION so that `#if defined(USE_FLASH_ATTENTION)` is true

Reply:
figured it out nvm